packages/automerge-repo-react-hooks/src/useBootstrap.ts

```diff
@@ -0,0 +1,58 @@
+import { useRepo } from "automerge-repo-react-hooks";
+import { DocumentId } from "automerge-repo";
+import { useEffect, useState } from "react";
+
+// Set URL hash
+export const setHash = (hash: string, pushState = false) =>
+  void history[pushState ? "pushState" : "replaceState"]("", "", "#" + hash);
+
+// Get current URL hash
+export const useHash = () => {
+  const [hashValue, setHashValue] = useState(window.location.hash);
+  useEffect(() => {
+    const handler = () => void setHashValue(window.location.hash);
+    window.addEventListener("hashchange", handler);
+    return () => void window.removeEventListener("hashchange", handler);
+  }, []);
+  return hashValue;
+};
+
+// Get a key from a query-param-style hash URL
+export const getQueryParamValue = (key: string, hash) =>
+  hash.match(new RegExp(`${key}=([^&]*)`))?.[1];
+
+export const createDocument = (repo, onCreate) => {
+  const handle = repo.create(); // Create a new document
+  if (onCreate) handle.change(onCreate); // Set initial state
+  return handle;
+};
+
+export const useBootstrap = ({
+  onCreate = () => {},
+  hashRouteKey = "id",
+  localStorageKey = "documentId",

```

Comment:
Setting this to `false` will disable localStorage (not yet implemented/tested).

Reply:
How about we make the key consistent between the localStorage and URL, default to documentId, and allow it to be passed in. What do you think?

Reply:
sure will do